lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AccessTokenService.java

```diff
@@ -1,83 +1,88 @@
 package uk.gov.di.ipv.cri.address.library.service;
 
 import com.nimbusds.common.contenttype.ContentType;
+import com.nimbusds.jwt.SignedJWT;
 import com.nimbusds.oauth2.sdk.AccessTokenResponse;
+import com.nimbusds.oauth2.sdk.AuthorizationCode;
 import com.nimbusds.oauth2.sdk.AuthorizationCodeGrant;
-import com.nimbusds.oauth2.sdk.GrantType;
 import com.nimbusds.oauth2.sdk.OAuth2Error;
 import com.nimbusds.oauth2.sdk.TokenRequest;
-import com.nimbusds.oauth2.sdk.TokenResponse;
+import com.nimbusds.oauth2.sdk.auth.ClientAuthentication;
+import com.nimbusds.oauth2.sdk.auth.PrivateKeyJWT;
 import com.nimbusds.oauth2.sdk.http.HTTPRequest;
+import com.nimbusds.oauth2.sdk.id.Audience;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import com.nimbusds.oauth2.sdk.id.Issuer;
+import com.nimbusds.oauth2.sdk.id.JWTID;
+import com.nimbusds.oauth2.sdk.id.Subject;
 import com.nimbusds.oauth2.sdk.token.AccessToken;
 import com.nimbusds.oauth2.sdk.token.BearerAccessToken;
 import com.nimbusds.oauth2.sdk.token.Tokens;
-import uk.gov.di.ipv.cri.address.library.exception.AccessTokenRequestException;
+import uk.gov.di.ipv.cri.address.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.cri.address.library.exception.AccessTokenValidationException;
+import uk.gov.di.ipv.cri.address.library.exception.ClientConfigurationException;
+import uk.gov.di.ipv.cri.address.library.exception.SessionValidationException;
 import uk.gov.di.ipv.cri.address.library.helpers.ListUtil;
 import uk.gov.di.ipv.cri.address.library.persistence.DataStore;
 import uk.gov.di.ipv.cri.address.library.persistence.item.AddressSessionItem;
 
 import java.net.URI;
-import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.UUID;
 
 public class AccessTokenService {
     public static final String CODE = "code";
-    public static final String REDIRECT_URI = "redirect_uri";
     public static final String GRANT_TYPE = "grant_type";
     public static final String CLIENT_ID = "client_id";
+    public static final String CLIENT_ASSERTION_TYPE = "client_assertion_type";
+    public static final String CLIENT_ASSERTION = "client_assertion";
+    public static final String AUTHORISATION_CODE = "authorization_code";
     private final DataStore<AddressSessionItem> dataStore;
     private final Long bearAccessTokenTtl;
-
-    public AccessTokenService(DataStore<AddressSessionItem> dataStore, Long bearerAccessTokenTtl) {
+    private final ConfigurationService configurationService;
+    private final JWTVerifier jwtVerifier;
+    private ListUtil listHelper = new ListUtil();
+
+    public AccessTokenService(
+            DataStore<AddressSessionItem> dataStore,
+            Long bearerAccessTokenTtl,
+            ConfigurationService configurationService,
```

Comment:
I think `bearerAccessTokenTtl` comes from `configurationService.getBearerAccessTokenTtl` at some stage this class only needed  `bearerAccessTokenTtl` hence there was no reason to pass in the entire configurationService. I think we can remove  `bearerAccessTokenTtl`

Reply:
Removed.